@navikt/core/react/src/dialog/popup/DialogPopupInternal.tsx

```diff
@@ -0,0 +1,274 @@
+      | (() => HTMLElement | null | undefined);
+    /**
+     * Will try to focus the given element on unmount.
+     *
+     * If not provided, Dialog will try to focus the `Dialog.Trigger` element
+     * or the last focused element before the dialog opened.
+     */
+    returnFocus?:
+      | React.RefObject<HTMLElement | null>
+      | (() => HTMLElement | null | undefined);
+
+    /**
+     * The position of the dialog relative to the viewport.
+     * @default "center"
+     */
+    position?: DialogPosition;
+    /**
+     * CSS `width`
+     *
+     * Has no effect when `position` is set to `fullscreen`.
+     *
+     * @default "medium"
+     */
+    width?: ResponsiveProp<string & {}> | "small" | "medium" | "large";
+    /**
+     * CSS `height`
+     *
+     * Has no effect when `position` is set to `fullscreen`, `left` or `right`.
+     */
+    height?: ResponsiveProp<string & {}> | "small" | "medium" | "large";
+    /**
+     * Adds a backdrop behind the dialog popup.
+     * @default true
+     */
+    withBackdrop?: boolean;
+  };
+
+/**
+ * @see 🏷️ {@link DialogPopupProps}
+ * @example
+ * ```jsx
+ * ```
+ */
+const DialogPopupInternal = forwardRef<
+  HTMLDivElement,
+  DialogPopupInternalProps
+>(
+  (
+    {
+      className,
+      modal = true,
+      closeOnOutsideClick = true,
+      initialFocus,
+      returnFocus,
+      position = "center",
+      width = "medium",
+      height,
+      id,
+      style,
+      "aria-labelledby": ariaLabelledbyProp,
+      withBackdrop,
+      ...restProps
+    },
+    forwardedRef,
+  ) => {
+    const { cn } = useRenameCSS();
+    const {
+      mounted,
+      popupRef,
+      setPopupElement,
+      triggerElement,
+      setOpen,
+      open,
+      transitionStatus,
+      popupElement,
+      nestedOpenDialogCount: nestedOpenDialogCountProp,
+      nested,
+      size,
+      titleId,
+      popupId,
+      onOpenChangeComplete,
+      setMounted,
+    } = useDialogContext();
+
+    const mergedRefs = useMergeRefs(forwardedRef, popupRef, setPopupElement);
+
+    useScrollLock({
+      enabled: open && modal === true,
+      mounted,
+      open,
+      referenceElement: popupElement,
+    });
+
+    /**
+     * On mount, popupRef is not defined in root, so we need to
+     * run hook here as well as root to ensure animations are tracked correctly
+     */
+    useOpenChangeAnimationComplete({
+      open,
+      ref: popupRef,
+      onComplete() {
+        if (open) {
+          onOpenChangeComplete?.(true);
+        } else {
+          setMounted(false);
+          onOpenChangeComplete?.(false);
+        }
+      },
+    });
+
+    const resolvedInitialFocus = initialFocus ?? popupRef;
+
+    const resolvedReturnFocus = () => {
+      if (returnFocus) {
+        return typeof returnFocus === "function"
+          ? returnFocus()
+          : returnFocus.current;
+      }
+
+      if (triggerElement?.checkVisibility()) {
+        return triggerElement ?? true;
```

Comment:
I don't think `triggerElement` can be nullish when `checkVisibility()` returned true 😅

```suggestion
        return triggerElement;
```